util/src/main/java/tech/pegasys/artemis/util/hashtree/HashTreeUtil.java

```diff
@@ -28,70 +28,54 @@
 public final class HashTreeUtil {
 
   /**
-   * Calculate the hash tree root of the provided value
+   * Create the hash tree root of a set of values of basic SSZ types or tuples of basic types. Basic
+   * SSZ types are uintN, bool, and byte. bytesN (i.e. Bytes32) is a tuple of basic types. NOTE:
+   * Bytes (and not Bytes32, Bytes48 etc.) IS NOT a basic type or a tuple of basic types.
    *
-   * @param value
+   * @param bytes One Bytes value or a list of homogeneous Bytes values.
+   * @return The SSZ tree root hash of the values.
+   * @see <a
+   *     href="https://github.com/ethereum/eth2.0-specs/blob/v0.5.1/specs/simple-serialize.md">SSZ
+   *     Spec v0.5.1</a>
    */
-  public static Bytes32 hash_tree_root(Bytes value) {
-    return SSZ.hashTreeRoot(value);
+  public static Bytes32 hash_tree_root_basic_type(Bytes... bytes) {
+    if (bytes.length == 1) {
+      if (bytes[0].size() > 32) {
+        return Hash.keccak256(bytes[0]);
```

Comment:
I'm not sure if this is in line with spec.

Reply:
Thanks, will fix this.